generators/chipyard/src/main/scala/upf/ChipTopUPF.scala

```diff
@@ -0,0 +1,71 @@
+// See LICENSE for license details
+package chipyard.upf
+
+import chipyard.{TestHarness, ChipTopLazyRawModuleImp, DigitalTop}
+import freechips.rocketchip.diplomacy.LazyModule
+import chisel.{UPFAspect, UPFFunc, ChiselUPFElement}
+
+import scala.collection.mutable.ListBuffer
+
+object ChipTopUPF {
+  def default: UPFFunc.UPFFunction = {
+    case top: ChipTopLazyRawModuleImp => {      
+      val modulesList = createModulesList(top)
+      var (g, pdList) = createPowerDomains(modulesList)
+      g = connectPDHierarchy(g, pdList)
+      g.bfsVisitor(g.rootObj, UPFGenerator.generateUPF)
+    }
+  }
+
+  def createModulesList(top: ChipTopLazyRawModuleImp): ListBuffer[LazyModule] = {
+    var modulesList = ListBuffer[LazyModule]()
+    modulesList.append(top.outer.lazySystem)
+    val tiles = top.outer.lazySystem match {
+      case t: DigitalTop => t.tiles.map(x => x)
+      case _ => throw new Exception("Unsupported BuildSystem type")
+    }
+    for (tile <- tiles) {
+      modulesList.append(tile)
+    }
```

Comment:
Why is this needed? Can't you just traverse the children modules to get the tiles?

Reply:
Oh do you mean get the tiles of each LazyModule in the traversal and add them to the overarching modules list? Or do you mean that the tiles should be in the traversal of getChildren? Because when trying out the traversal, I didn't see any tiles in the resulting modules list so I had to manually add them as such.

Reply:
Changed my usage of getChildren to travel the full hierarchy, now it's giving me the tiles.